sway-core/src/parse_tree/code_block.rs

```diff
@@ -58,55 +58,111 @@ impl CodeBlock {
                         warnings,
                         errors
                     );
-                    vec![AstNode {
-                        content: AstNodeContent::Expression(evaluated_node),
+                    let mut ast_node_contents = evaluated_node_result
+                        .var_decls
+                        .into_iter()
```

Comment:
I see `parse_result.var_decls.into_iter().map(func).collect()` a _lot_ here -- what do you think about exposing an API that looks something like this?

```rust
parse_result.collect_vars(func)
```

?

That could save some LOC and provide a more expressive API over `ParseResult`